Refactor FxGraphCache.load into separate functions, so that AOTAutogradCache may access it correctly later

jamesjwu opened the pull request:
Summary:
We refactor FxGraphCache.load into three phases:
- prepare_key, which checks that an inductor input is cacheable and bypasses otherwise
- load_with_key, which tries to lookup the key in the cache
- post compile, where we do some logging and run post compile steps

Splitting it along these lines will allow AOTAutogradCache to use load_with_key and still get access to all of the observability + remote cache logic when accessing FxGraphCache, without needing to pass key components, etc.

Differential Revision: D62314862


cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

oulgen commented on torch/_inductor/codecache.py:
```diff
@@ -1274,6 +1274,109 @@ def _check_can_cache(gm: torch.fx.GraphModule) -> None:
+        except ModuleNotFoundError as e:
+            # No need for a stack trace on this error
+            remote_cache = None
+            log.warning("Unable to create a remote cache: %s", e)
```
why do you have two except blocks that do literally the same?

jamesjwu replied:
This is just the previous behavior moved into a new function, see lines below.

I think @aorenste added it in 
https://github.com/pytorch/pytorch/commit/7239b8a4f1ab99f3b94c87342720db88a1660626 added this so that there's no stack trace for the module not found error

jamesjwu commented:
^ The cancelled jobs already passed in an earlier version of this PR so I will just land

jamesjwu committed:
Refactor FxGraphCache.load into separate functions, so that AOTAutogradCache may access it correctly later (#135491)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/135491

We refactor FxGraphCache.load into three phases:
- prepare_key, which checks that an inductor input is cacheable and bypasses otherwise
- load_with_key, which tries to lookup the key in the cache
- post compile, where we do some logging and run post compile steps

Splitting it along these lines will allow AOTAutogradCache to use load_with_key and still get access to all of the observability + remote cache logic when accessing FxGraphCache, without needing to pass key components, etc.

Test Plan:
Existing tests should pass, fully a refactor
Ran benchmark as a integration test and see that fx graph cache does populate properly.

Cache miss:
https://interncache-all.fbcdn.net/manifold/tlparse_reports/tree/logs/jjwu/custom/miss/index.html

Cache hit:
https://interncache-all.fbcdn.net/manifold/tlparse_reports/tree/logs/jjwu/custom/hit/index.html

Cache bypass:
https://interncache-all.fbcdn.net/manifold/tlparse_reports/tree/logs/jjwu/custom/bypass/index.html

Results in remote fx graph cache scuba table:
https://fburl.com/scuba/pt2_remote_cache/715v52ex

Reviewed By: oulgen

Differential Revision: D62314862
